extensions/kotlin-serialization/runtime/src/main/kotlin/io/quarkus/kotlin/serialization/KotSerMessageBodyReader.kt

```diff
@@ -0,0 +1,50 @@
+package io.quarkus.kotlin.serialization
+
+import io.quarkus.resteasy.reactive.server.runtime.StreamUtil
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.serializer
+import org.jboss.resteasy.reactive.common.providers.serialisers.AbstractJsonMessageBodyReader
+import org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo
+import org.jboss.resteasy.reactive.server.spi.ServerMessageBodyReader
+import org.jboss.resteasy.reactive.server.spi.ServerRequestContext
+import java.io.InputStream
+import java.lang.reflect.Type
+import javax.inject.Inject
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.MultivaluedMap
+
+class KotSerMessageBodyReader : AbstractJsonMessageBodyReader(), ServerMessageBodyReader<Any> {
```

Comment:
I don't understand this one.

Reply:
The Maven module name seems to imply that this module is general serialization module, something like `quarkus-jackson`, when it in fact is specific to RESTEasy Reactive, so it should have resteasy-reactive in the name (or think about whether it makes sense to have 2 modules)

Reply:
oh, i see.  hrm.  you're probably right.  `quarkus-resteasy-reactive-kotlin-serialization` is definitely a mouthful, though.  It actually might be pretty general because it only really just registers those message body reader/writer impls.  there's not really a direct reference to resteasy-reactive specifically.  Maybe once the client side is done this distinction will be clearer?  Should we hold off deciding until then?

Reply:
We should have a reference to RESTEasy Reactive for consistency reasons - users are used to what we are doing with Jackson and JSON-B

Reply:
What would the client support look like? Would it reuse the same Writer and Reader?

Reply:
I would expect so.

Reply:
Let's see how that shapes then. In any case, the name of the module will need to be changed.

Reply:
module renamed and moved under the `resteasy-reactive` directory with the other RR extensions.

Reply:
OK thanks. I'd prefer to see the client stuff as well since the code will likely will be shared.

Reply:
I was hoping to do it in two smaller PRs since I've never mucked with the client side before (Stuart got there first which was great).  But if you'd prefer one big one, I can withdraw this one and start on the client.  At a minimum, I've validated that the server side changes are correct with this PR which was half the motivation for separate PRs.

Reply:
The main reason is that I want to see what the modules will look like.
I am guessing that the Reader and Writer for the client will be exactly the same as for the server, which means that some care must be taken.
FWIW, the client module should be simple to do now that you have the server done.

Reply:
That's fair.  i'll withdraw this one and start on the client, then.